noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/note.nr

```diff
@@ -34,14 +37,9 @@ where
 
     let mut fields = [0; N + 2];
 
-    // Note that we're almost accidentally following the standard log encoding here: because the note type id only uses
-    // 7 bits, it just barely fits in the log metadata, and the log type id is implicitly 0 (i.e. a private note log).
-    // Partial notes modify `get_id` to have it set the 8th bit, resulting in a log type id of 1 (i.e. a partial note
-    // private log). Fields 1 to len are the note content, which we here hardcode to be the storage slot in the first
-    // field, and the packed note following after.
-    // Ideally we'd know if this is a private note or a partial note, and call a function that'd be the opposite of
-    // discovery::private_notes::decode_private_note_log (or partial accordingly).
-    fields[0] = Note::get_id();
+    // We pack log type id and log metadata into the first field. Search for `decode_log_plaintext` function to see
+    // where the value gets decoded.
+    fields[0] = (Note::get_id() * 2.pow_32(64)) + log_type_id as Field;
```

Comment:
I'd consider creating a function that takes a log type id and log metadata and creates the expanded metadata, i.e. the inverse of the one I sugested before. We'll repeat this snippet for events etc, and it'd be nice to have some simple tests for it.

Reply:
Good idea. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/12956/commits/bde6715045a08cabfdf77cfb743d5f1c9daee32c